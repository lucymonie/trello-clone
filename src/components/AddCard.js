import React, { Component } from 'react';
import SaveCancel from './SaveCancel';
import { addNewItem, getData } from '../helpers/database';

class AddCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showAdd: false,
      textEntered: ""
    }
  }

  componentDidMount() {
    this.updateTasks()
  }

  addNewTask = (taskText) => {
    addNewItem("tasks", {"task_text": taskText});
    this.setState({ textEntered: "", showAdd: false });
    this.updateTasks();
  }

  updateTasks = () => {
    getData("tasks").then((tasksData) => {
      this.setState({ tasksData });
    });
  }

  render() {
    const { showAdd, textEntered } = this.state;
      return (
      <div>
        {!showAdd &&
        <div className="add-card grey-text cursor-pointer"
          onClick={ () => this.setState({ showAdd: true }) }
        >Add a card...</div>}
        {showAdd &&
        <div>
          <textarea className="card-input white"
            rows={3}
            onChange={ (e) => this.setState({textEntered: e.target.value} )}
            value={textEntered}
          />
          <SaveCancel
            onClickCancel={() => this.setState({ textEntered: "", showAdd: false }) }
            onClickSave={this.addNewTask} />
        </div>
        }
      </div>
    )
  }
}

export default AddCard;
