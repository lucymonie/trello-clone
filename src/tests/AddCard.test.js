import React from 'react';
import { shallow } from 'enzyme';
import AddCard from '../components/AddCard';


describe("AddCard", () => {
  it("renders consistently", () => {
    const component = shallow(<AddCard updateTasks={jest.fn()} />);
    expect(component).toMatchSnapshot();
  });

  it("should render 'Add a card...' if showAdd is false", () => {
    const component = shallow(<AddCard updateTasks={jest.fn()} showAdd={false} />);
    const firstDiv = component.find("div").at(0);
    expect(firstDiv.text()).toBe("Add a card...");
  });

  it("should render a textarea if showAdd is true", () => {
    const mockEvent = jest.fn();
    const component = shallow(<AddCard updateTasks={mockEvent} showAdd={true} />);
    const textarea = component.find("textarea");
    expect(textarea).toMatchSnapshot();
  });

  describe("when clicking the Add a card div", () => {
    it("triggers the onClick", () => {
      const mockEvent = jest.fn();
      const component = shallow(<AddCard updateTasks={mockEvent} showAdd={false} />);
      const firstDiv = component.find("div").at(0);
      firstDiv.simulate("click");
      expect(mockEvent.mock.calls.length).toEqual(1);
    });
  });

});
