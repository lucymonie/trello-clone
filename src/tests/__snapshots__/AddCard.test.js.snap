// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCard renders consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddCard
    updateTasks={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="add-card grey-text cursor-pointer"
          onClick={[Function]}
>
          Add a card...
</div>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add a card...",
          "className": "add-card grey-text cursor-pointer",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add a card...",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="add-card grey-text cursor-pointer"
            onClick={[Function]}
>
            Add a card...
</div>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add a card...",
            "className": "add-card grey-text cursor-pointer",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add a card...",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`AddCard should render a textarea if showAdd is true 1`] = `
ShallowWrapper {
  "length": 0,
  Symbol(enzyme.__root__): ShallowWrapper {
    "length": 1,
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <AddCard
      showAdd={true}
      updateTasks={[Function]}
/>,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="add-card grey-text cursor-pointer"
            onClick={[Function]}
>
            Add a card...
</div>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add a card...",
            "className": "add-card grey-text cursor-pointer",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add a card...",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="add-card grey-text cursor-pointer"
              onClick={[Function]}
>
              Add a card...
</div>,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add a card...",
              "className": "add-card grey-text cursor-pointer",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add a card...",
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
